lib/Differentiator/DiffPlanner.cpp

```diff
@@ -14,6 +14,53 @@ using namespace clang;
 namespace clad {
   static SourceLocation noLoc;
 
+  // Returns address of parent of Expr node obtained 
+  // by skipping past any parantheses, implicit casts
+  // and UnaryOperators from Expr pointed by `E`.
+  // If considerParanAsParent is false,then returns last non-parantheses
+  // ancestor of Expr node obtained by skipping.
+  // Returns nullptr, if there is no skipping of nodes
+  // from given Expr
+  Expr* getParentOfSkipPastParImpCastUnOp(Expr* E,bool considerParanAsParent = true) {
+    Expr* last_E = nullptr;
+    while(1) {
+      if(auto ICE = dyn_cast<ImplicitCastExpr>(E)) {
```

Comment:
```suggestion
      if (auto ICE = dyn_cast<ImplicitCastExpr>(E)) {
```
And in the rest of the changes.

Reply:
I will take care of the spaces. 
Can we use some formatter to take care of formatting issues in github workflows ?

Reply:
We can and we should, if you have experience with that I'd be happy to merge your PR (cc @reikdas).

Reply:
I don't have much experience with this yet, I will try to do it, if I succeed I will send the PR :)